src/main/java/com/google/devtools/build/lib/skyframe/WorkspaceFileFunction.java

```diff
@@ -100,27 +101,49 @@ public SkyValue compute(SkyKey skyKey, Environment env)
       return null;
     }
 
-    Optional<RootedPath> resolvedFile =
-        RepositoryDelegatorFunction.RESOLVED_FILE_INSTEAD_OF_WORKSPACE.get(env);
-    if (resolvedFile == null) {
-      return null;
+    // The final content of the WORKSPACE is calculated in the following ways:
+    // 1. If --resolved_file_instead_of_workspace is enabled, the final content will be:
+    //    getDefaultWorkspacePrefix() + workspaceFromResolvedValue().
+    // 2. Otherwise, if --experimental_enable_bzlmod is enabled and WORKSPACE.bzlmod exists,
+    //    the final content will be:
+    //    WORKSPACE.bzlmod (Neither of prefix or suffix are added)
+    // 3. Otherwise, the final content will be:
+    //    getDefaultWorkspacePrefix() + WORKSPACE + getDefaultWorkspaceSuffix()
+
+    Optional<RootedPath> resolvedFile = Preconditions.checkNotNull(
+        RepositoryDelegatorFunction.RESOLVED_FILE_INSTEAD_OF_WORKSPACE.get(env));
+    boolean useWorkspaceResolvedFile = resolvedFile.isPresent();
+
+    boolean useWorkspaceBzlmodFile = false;
+    RootedPath workspaceBzlmodFile = RootedPath.toRootedPath(workspaceFile.getRoot(),
+        workspaceFile.getRootRelativePath().replaceName("WORKSPACE.bzlmod"));
+    // We only need to check WORKSPACE.bzlmod when the resolved file isn't used.
+    if (!useWorkspaceResolvedFile) {
```

Comment:
this condition can be simply rewritten as:

```java
if (!useWorkspaceResolvedFile && RepositoryDelegatorFunction.ENABLE_BZLMOD.get(env)) {
  FileValue ... = ...;
  if (...) { ... }
  useWorkspaceBzlmodFile = ...;
}
```

IMO the logic is much clearer that way.

Reply:
Oh, absolutely,, thanks!